File: util/Resize.m
Platform;

ifSeg = 1;
if ifSeg
    inDir = fullfile(dataDir, 'CarlaGen/C20_S1/Seg-1024');
else
    inDir = fullfile(dataDir, 'CarlaGen/C20_S1/RGB-1024');
end
outDir = mkdir2([inDir(1:end-4) '512']);

splits = {'train', 'val', 'test'};

for s = 1:length(splits)
    split = splits{s};
    inDirSplit = fullfile(inDir, split);
    outDirSplit = mkdir2(fullfile(outDir, split));
    dirList = dir(fullfile(inDirSplit, '*.png'));
    parfor i = 1:length(dirList)
        if ifSeg
            [idx, cm] = imread(fullfile(inDirSplit, dirList(i).name));
            idx = imresize(idx, [512 1024], 'nearest');
            imwrite(idx, cm, fullfile(outDirSplit, dirList(i).name));
        else
            img = imread(fullfile(inDirSplit, dirList(i).name));
            img = imresize(img, [512 1024]);
            imwrite(img, fullfile(outDirSplit, dirList(i).name));
        end
    end
end

File: SemSeg/deeplab/deeplab/utils/resize.py
from os.path import join, isdir, basename
from os import makedirs
from glob import glob
from PIL import Image

data_dir = 'Data'
in_dir = join(data_dir, 'CarlaGen/C20_S1/RGB-1024')
out_dir = join(data_dir, 'CarlaGen/C20_S1/RGB-512')

splits = ['train', 'val', 'test']

for s in splits:
    in_dir_split = join(in_dir, s)
    out_dir_split = join(out_dir, s)
    if not isdir(out_dir_split):
        makedirs(out_dir_split)
        
    files = glob(join(in_dir_split, '*.png'))

    for i, f in enumerate(files):
        print('Processing %d/%d - %s' % (i, len(files), f))
        img = Image.open(f)
        img = img.resize((1024, 512), Image.BICUBIC)
        img.save(join(out_dir_split, basename(f)))
    